js/content/community.js

```diff
@@ -1,27 +1,94 @@
 
 let SteamId = (function(){
+    function SteamId(steam64str) {
+        this._steamId64 = steam64str || this.getSteamId64();
+        this._steamId = this.fromString(this._steamId64);
+    };
 
-    let self = {};
-    let _steamId = null;
+    function divide(str) {
+        let length = str.length;
+        let result = [];
+        let num = 0;
+        for (let i=0; i<length; i++) {
+            num += Number(str[i]);
+
+            let r = Math.floor(num / 2);
+            num = ((num - 2*r) * 10);
+
+            if (r != 0 || result.length != 0) {
+                result.push(r);
+            }
+        }
+
+
+        return [result, num > 0 ? 1 : 0];
+    }
+
+    function getBinary(str) {
+        let upper32 = 0;
+        let lower32 = 0;
+
+        let index = 0;
+        let bit = 0;
+        do {
+            [str, bit] = divide(str);
+            
+            if (bit) {
+                if (index < 32) {
+                lower32 = lower32 | (1 << index);
+                } else {
+                    upper32 = upper32 | (1 << (index - 32));
+                }
+            }
+            
+            index++;
+        } while(str.length > 0);
+
+        return [upper32, lower32];
+    }
+
+    SteamId.prototype.fromString = function(steam64str) {
+        let [upper32, lower32] = getBinary(steam64str);
+        return {
+            y: lower32 & 1,
+            accountNumber: (lower32 & ((1 << 31) - 1) << 1) >> 1,
+            instance: (upper32 & ((1 << 20) - 1)),
+            type: (upper32 & (15 << 20)) >> 20,
+            universe: (upper32 & ((1 << 8) - 1) << 20) >> 20
+        }
+    };
+
+    SteamId.prototype.getAccountId = function() {
+        return this._steamId.accountNumber * 2;
+    };
 
-    self.getSteamId = function() {
-        if (_steamId) { return _steamId; }
+    SteamId.prototype.getSteamId2 = function() {
+        return "STEAM_0:0:" + this._steamId.accountNumber;
+    };
+    
+
+    SteamId.prototype.getSteamId3 = function() {
+        return "[U:1:" + this.getAccountId() + "]";
+    };
+
+    SteamId.prototype.getSteamId64 = function() {
+        if (this._steamId64) { return this._steamId64; }
 
         if (document.querySelector("#reportAbuseModal")) {
-            _steamId = document.querySelector("input[name=abuseID]").value;
+            this._steamId64 = document.querySelector("input[name=abuseID]").value;
```

Comment:
I think it might be better to make SteamId a standalone class just handling data, and handle DOM/source of IDs either in place where it's needed, or in another class. That way could use SteamId as a normal object, similar to how Price() is used

Reply:
We could then use this object everywhere where we use just Steam64 string